src/Federation/Resolvers/BaseBatchedReferenceResolver.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+
+namespace Nuwave\Lighthouse\Federation\Resolvers;
+
+use GraphQL\Type\Introspection;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Nuwave\Lighthouse\Federation\BatchedEntityResolver;
+
+abstract class BaseBatchedReferenceResolver implements BatchedEntityResolver
```

Comment:
Let's hold off on this for this pull request. Feel free to propose it separately.